src/reference/asciidoc/router.adoc

```diff
@@ -589,6 +589,35 @@ The example below demonstrates a `PayloadTypeRouter` configuration which is equi
 </int:payload-type-router>
 ----
 
+The equivalent router, using Java configuration:
+
+[source, java]
+----
+@ServiceActivator(inputChannel = "routingChannel")
+@Bean
+public PayloadTypeRouter router() {
+    PayloadTypeRouter router = new PayloadTypeRouter();
+    router.setChannelMapping(String.class.getName(), "stringChannel");
+    router.setChannelMapping(Integer.class.getName(), "integerChannel");
+    return router;
+}
+----
+
+and using the Java DSL:
+
+[source, java]
+----
+@Bean
+public IntegrationFlow routerFlow() {
+    return IntegrationFlows.from("routingChannel")
+            .route(new PayloadTypeRouter(), m -> m
```

Comment:
Yes, I think show both options would be great.
That will be some kind of sample which end users are always seeking.
So, let's move on with the `new PayloadTypeRouter()` (but proper), and regular, based on the Lambda and generics resolution during typing.